pulsar-common/src/main/java/org/apache/pulsar/client/admin/internal/data/AuthPoliciesImpl.java

```diff
@@ -42,6 +42,10 @@ public final class AuthPoliciesImpl implements AuthPolicies {
     @JsonProperty("subscription_auth_roles")
     private Map<String, Set<String>> subscriptionAuthentication = new TreeMap<>();
 
+    // Default value is set in the builder
+    @JsonProperty(value = "implicit_subscription_auth")
+    private boolean implicitSubscriptionAuth;
```

Comment:
adding a field with default value "true" is problematic, thinking to the upgrade path.
This is because on storage (ZK) all the old Policies do not have this field, but we are considering it "true" if it is not present.
This may make us fall into some subtle problems with clusters upgraded to the latest version.

I suggest to go down these ways:
1) use Boolean (nullable boolean) and deal explicitly with the case "unknown"
2) negate the flag, this way the default will be "false" and this value will also be applied in case of null/non existing field (so old cluster upgraded, or updates happening during rolling upgrades of the cluster

I prefer 2) as it is easier to deal with both code changes and also with the migration

Reply:
Good points Enrico. Yes this should be addressed.

Reply:
Thank you for the review, @eolivelli. My preference is also 2. I'll replace `implicit` with `explicit` for the whole PR/PIP.

Reply:
@eolivelli - on second thought, are you suggesting that I need to update the variable names or just that the stored metadata needs to default to false? The semantics of "granting permission" on the API will be different if we change the endpoint names to align with a default value where "false" means additional permission is granted roles. For example, an alternate name for the `/implicitPermissionOnSubscription` endpoint is `/permissionOnSubscriptionRequired`. In that case, granting permission would use a `PUT`/`POST` to set the value to false, which seems counter intuitive.

Reply:
The main point is that the default value must be 'false' or null because this is what is on existing data.

Reply:
That makes sense. I think the likely alternative name is `permissionOnSubscriptionRequired`. If someone has a better name, though, I'll use that.

Reply:
@eolivelli - I updating the naming by using `permissionOnSubscriptionRequired`. Please let me know what you think about the name.

Reply:
works for me